### DISCUSSION TITLE
Possible collaboration between GONNX and gomlx?

### DISCUSSION THREAD

**Discussion by Swopper050** [gomlx/onnx-gomlx#3] [General]
Hi there @janpfeifer! 

Awesome work on `gomlx`, amazing! I just discovered your package a few days ago.
Some time ago, together with a colleague of mine we built [gonnx](https://github.com/AdvancedClimateSystems/gonnx).
We were missing a package that could run our ONNX models in Go (with GRU and LSTM layers). Gonnx is by no means complete, and still missing operators and different operator sets, but I thought maybe it is interesting to look if there is a way of combining the packages, or in some way share implementation?

Our backend currently is build upon [Tensor](`https://github.com/gorgonia/tensor`).

What would this package mostly need now? Maybe we could try combining some efforts? What do you think?

**Comment by janpfeifer** [comment]
hey Bram, I hadn't seen `gonnx` looks very cool!

Funny that one of my next projects is going to add Gorgonia's Tensor library as a backend for GoMLX :) While XLA is immensely faster (desirable when training or larger models), it has this huge inconvenience of having C++ libraries that needs installing, and it doesn't support Darwin (it's broken/missing pieces). So I urgently need to add a GoMLX backend using Tensor.

But one of the main reasons we (with folks from [KnightsAnalytics](https://www.knightsanalytics.com/) conceived `onnx-gomlx` was to be able to import a ONNX model and then further fine-tune them -- GoMLX's autodiff/training support makes that relatively easy.

Hey, you be up for a sync up ? Maybe we could brainstorm where to collaborate ?

**Reply by Swopper050** [reply]
Yeah sure! I'd also contacted @wipsel (co-author of Gonnx) and he was also open for a brainstorm, I'll drop you an email.
